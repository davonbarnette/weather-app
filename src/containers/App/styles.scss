@import "../../styles/global";

.app {

  @include flex-common(column, center, center);
  box-sizing: border-box;
  padding:24px;
  min-height:100vh;

  &-content {
    max-width:800px;
    width:100%;

    .data {
      margin-top:24px;
      box-shadow: (0 7px 14px rgba(50, 50, 93, .1), 0 3px 6px rgba(0, 0, 0, .03));
      border: 1px solid #e0e0e0;
      border-radius:5px;

      &-title {
        font-weight:600;
        color: $dark-gray;
        margin-bottom: 18px;
      }

      .current-weather-wrapper {
        border: 1px solid #e0e0e0;
        border-style: none none solid none;
        padding:24px;
      }

      .forecast-wrapper {
        padding:24px;
      }
    }

    .get-forecast {
      padding: 12px 18px;
      border:none;
      border-radius:3px;
      background:#13b369;
      color:white;
      font-weight:600;
      transition: all 0.2s;
      outline:none;
      &:not(.loading):hover {
        cursor: pointer;
        transform: translate(0, -3px);
      }
      &.loading {
        background:#8e8e8e;
      }
      &:active {
        border:none;
      }
    }

    &-subtitle {
      text-transform: uppercase;
      color:$light-gray;
      font-weight:600;
      letter-spacing: 0.15em;
      font-size:13px;
    }
    &-title {
      font-weight:700;
      color:$dark-gray;
      font-size:36px;
      margin-bottom:12px;
      line-height:1em;
    }
    &-description {
      @include flex-common(column, flex-start, flex-start);
      font-size:14px;
      color: $dark-gray;
      margin-bottom:24px;
      font-weight:500;
      .code {
        padding: 6px 12px;
        background:#cecece;
        border-radius:5px;
        border: 1px solid #9c9c9c;
        font-weight:600;
        margin-top:6px;
      }
    }
  }
}