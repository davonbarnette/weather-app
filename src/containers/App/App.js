import React, {useState} from 'react';
import {observer} from "mobx-react";
import 'moment-timezone';
import cx from 'classnames';

import './styles.scss';

import {AppActions} from "../../Actions";
import AppStore from '../../Store';
import CurrentWeather from "../../components/CurrentWeather/CurrentWeather";
import Forecast from "../../components/Forecast/Forecast";

AppStore.init();

const App = (props) => {

    const [currentWeather, setCurrentWeather] = useState(undefined);
    const [forecast, setForecast] = useState(undefined);
    const [loading, setLoading] = useState(false);

    async function getAllWeather(){
        if (loading) return null;

        setLoading(true);
        let location = await AppActions.attemptToGetCurrentLocation();
        if (location){
            const {latitude, longitude} = location;
            let weatherResponse = await AppActions.getCurrentWeather(latitude, longitude);
            setCurrentWeather(weatherResponse);

            let forecastResponse = await AppActions.get5DayForecast(latitude, longitude);
            setForecast(forecastResponse);
        }
        setLoading(false);
    }

    return (
        <div className="app">
            <div className='app-content'>
                <div className='app-content-subtitle'>
                    Welcome To
                </div>
                <div className='app-content-title'>
                    Davon's Weather
                </div>
                <div className='app-content-description'>
                    <div className='built-with'>
                        This was built with:
                        <ul>
                            <li>React</li>
                            <li>MobX</li>
                        </ul>
                    </div>
                    <div>This app will give you the current weather, and your 5-day forecast as well. If you're running
                        this on the development server, you'll need to create a .env file in your project's root directory
                        that has one entry:</div>
                    <div className='code'>REACT_APP_OPEN_WEATHER_API_KEY=REPLACE_WITH_YOUR_API_KEY</div>
                </div>
                <button className={cx('get-forecast', {loading})} onClick={getAllWeather}>
                    {loading ? 'Loading Weather Data...' : 'Get Forecast'}
                </button>
                {currentWeather && forecast &&
                <div className='data'>
                    <div className='current-weather-wrapper'>
                        <div className='data-title'>Current Weather</div>
                        <CurrentWeather currentWeather={currentWeather}/>
                    </div>
                    <div className='forecast-wrapper'>
                        <div className='data-title'>5-Day Forecast</div>
                        <Forecast forecast={forecast}/>
                    </div>
                </div>}
            </div>
        </div>
    );
}

export default observer(App);
