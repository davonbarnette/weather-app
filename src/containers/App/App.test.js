import React from 'react';
import { render } from '@testing-library/react';
import 'mobx-react-lite/batchingForReactDom'
import App from './App';

describe('App Component Renders', () => {
  test('Snapshot matches', () => {
    const {container} = render(<App/>);
    expect(container.firstChild).toMatchSnapshot();
  });
})
