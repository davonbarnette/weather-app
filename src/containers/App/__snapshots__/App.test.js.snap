// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component Renders Snapshot matches 1`] = `
<div
  class="app"
>
  <div
    class="app-content"
  >
    <div
      class="app-content-subtitle"
    >
      Welcome To
    </div>
    <div
      class="app-content-title"
    >
      Davon's Weather
    </div>
    <div
      class="app-content-description"
    >
      <div
        class="built-with"
      >
        This was built with:
        <ul>
          <li>
            React
          </li>
          <li>
            MobX
          </li>
        </ul>
      </div>
      <div>
        This app will give you the current weather, and your 5-day forecast as well. If you're running this on the development server, you'll need to create a .env file in your project's root directory that has one entry:
      </div>
      <div
        class="code"
      >
        REACT_APP_OPEN_WEATHER_API_KEY=REPLACE_WITH_YOUR_API_KEY
      </div>
    </div>
    <button
      class="get-forecast"
    >
      Get Forecast
    </button>
  </div>
</div>
`;
